packages/wallet-management/src/syncWagmiConfig.ts

```diff
@@ -13,13 +13,26 @@ export const syncWagmiConfig = async (
     _mainnet.contracts = { ...mainnet.contracts, ..._mainnet.contracts }
   }
   wagmiConfig._internal.chains.setState(chains)
-  wagmiConfig._internal.connectors.setState(() =>
-    [
+
+  wagmiConfig._internal.connectors.setState((currentConnectors) => {
+    const mipdProviders = wagmiConfig._internal.mipd?.getProviders() || []
+    const newMipdRdnsSet = new Set(
+      mipdProviders.map((provider) => provider.info.rdns)
+    )
+
+    const preservedConnectors = currentConnectors.filter((connector) => {
+      return !newMipdRdnsSet.has(connector.id)
+    })
+
+    const newConnectors = [
       ...connectors,
       ...(wagmiConfig._internal.mipd
         ?.getProviders()
         .map(wagmiConfig._internal.connectors.providerDetailToConnector) ?? []),
     ].map(wagmiConfig._internal.connectors.setup)
```

Comment:
I think we can reuse `mipdProviders` instead of getting them a second time here.